### DISCUSSION TITLE
[2023-10-23] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#8128] [Weekly Demo Day]
- James: Expand capabilities of `Table.set` and better dropdown support, #8005
- James: Integrating Enso Cloud with the libraries (part 1...) #8006
- James: Use US locale for Date/Time parsing by default. #8053
- James: Fixes for Table viz #8102
- Radosław: Added benchmarks related to `add_row_number` performance investigation, and created followup tickets
  #8091
- Radosław: Reviewed places where we can use `Column_Ref` - added to `replace`, `fill_nothing` and `fill_missing`
  #8101
- Radosław: Added `Previous_Value` mode in `fill_nothing` and `fill_missing` 
  #8105
- Radosław: Tested and improved handling of `Date_Time with_timezone=False` in Postgres connector 
  #8114
- Radosław: Fixed an incremental compilation issue related to SPI that was causing spurious CI failures 
  #8129 
- Greg: Add Table.expand_to_rows to allow flattening vector and array values in table #8042
- Greg: Restructuring XML into Table form #8083
- Greg: Pass problem aggregator in expand_to_rows #8103
- Greg: Overhaul the random number and item generation code #8127
- Jaroslav: PoC of the IDE working directly with Java types #8009
- Jaroslav: Avoid Null receiver values are not supported by libraries #8044
- Jaroslav: Icon and documentation for Enso VSCode extension #8051
- Jaroslav: Avoid null values in constant node #8061
- Jaroslav: Simplify code to run in the browser #8092
- Jaroslav: PoC: Running Enso parser in the browser #8094
- Jaroslav: Presenting [Enso at GeeCON](https://docs.google.com/presentation/d/1Vdy62v12noGaWtdZIhcV6eMtCwZoeCUtnanCVpisXYs), Prague
- Hubert: Eliminate VCS TimeoutExceptions on startup #8080
- Hubert: Ambiguous imports warnings/errors are serializable #8093
- Hubert: More debug info for failed edits #8107
- Hubert: Report human-readable names for shadowed parameters #8115
- Hubert: Investigating new GUI integration problems #8133 
- Pavel: Upgrade enso to GraalVM for jdk 21 #7991
- Dmitry: Suppress futile visualization warnings #8056
- Dmitry: Initialize suggestions database only once #8116

**Comment by radeusgd** [comment]
### `fill_nothing`

![image](https://github.com/enso-org/enso/assets/1436948/a86d1808-5423-4f37-99a5-0a9568adedff)
![image](https://github.com/enso-org/enso/assets/1436948/54126624-f6eb-4373-b3e3-83ee7e8c6fd6)

### `fill_empty`

![image](https://github.com/enso-org/enso/assets/1436948/c169641a-e358-47a5-9eba-fca6d8182bc9)
![image](https://github.com/enso-org/enso/assets/1436948/71758514-ab84-4185-bf4a-bff59c7e1eb2)

### `replace`

![image](https://github.com/enso-org/enso/assets/1436948/f03fe5fc-e45f-431b-8c3a-94e8f84eaaac)
 
![image](https://github.com/enso-org/enso/assets/1436948/19a74be4-3032-4059-acca-e8dbcf9f6cc4)
